### PR TITLE
Move parts of state-history-plugin to libraries/state_history

### PR DESCRIPTION
**Pull request by tbfleming**
## Change Description

This moves some parts of ship's logic into a library so the newest new CDT tester may include it.

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/state_history/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+             trace_converter.cpp
+             ${HEADERS}
+           )
+
+target_link_libraries( state_history 
+                       PUBLIC eosio_chain fc chainbase softfloat
+                     )
+
+target_include_directories( state_history
+                            PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_SOURCE_DIR}/../wasm-jit/Include"
```
Comment (arhag) [inline libraries/state_history/CMakeLists.txt:16]: It doesn't seem you actually need `"${CMAKE_CURRENT_SOURCE_DIR}/../wasm-jit/Include"` here.

File: libraries/state_history/trace_converter.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include <eosio/state_history/trace_converter.hpp>
+
+namespace eosio {
+namespace state_history {
+
+using eosio::chain::packed_transaction;
+using eosio::chain::plugin_exception;
+
+bool is_onblock(const transaction_trace_ptr& p) {
+   if (p->action_traces.size() != 1)
```
Comment (arhag) [inline libraries/state_history/trace_converter.cpp:11]: Should be:
```cpp
if (p->action_traces.empty())
```

The `is_onblock` in `trace_api/chain_extraction.hpp` also needs a similar correction.

**Comment by gleehokie** [comment]
@tbfleming should the new directory be named libraries/state-history instead of libraries/state_history (dash instead of underscore)?  We use "-" as the separator for directory names in the eos repo currently.

**Comment by tbfleming** [comment]
@gleehokie `state-history` isn't a valid namespace name and I thought it'd get confusing to have a directory name subtly different.

File: libraries/state_history/CMakeLists.txt (hunk 1 above)
Comment (tbfleming) [reply libraries/state_history/CMakeLists.txt]: removed

File: libraries/state_history/trace_converter.cpp (hunk 2 above)
Comment (tbfleming) [reply libraries/state_history/trace_converter.cpp]: changed

**Pull request merged**